### PR TITLE
Add missing type hints to tests.

### PR DESCRIPTION
**Pull request by clokep**
Handles `tests.metrics` and `tests.crypto`.

### CODE REVIEW

File: tests/crypto/test_keyring.py [hunk 1]
```diff
@@ -271,15 +284,15 @@ def test_verify_json_for_server_with_null_valid_until_ms(self):
             self.hs, key_fetchers=(StoreKeyFetcher(self.hs), mock_fetcher)
         )
 
-        key1 = signedjson.key.generate_signing_key(1)
+        key1 = signedjson.key.generate_signing_key("1")
         r = self.hs.get_datastores().main.store_server_verify_keys(
             "server9",
             time.time() * 1000,
-            [("server9", get_key_id(key1), FetchKeyResult(get_verify_key(key1), None))],
+            [("server9", get_key_id(key1), FetchKeyResult(get_verify_key(key1), None))],  # type: ignore[arg-type]
```
Comment (clokep) [inline tests/crypto/test_keyring.py:291]: I'm skeptical of this is `None` a valid value here or are we just abusing the API in order to insert the data to check how we deal with legacy data or something?

Comment (reivilibre) [reply tests/crypto/test_keyring.py:291]: I think you're right — we're abusing this here to insert legacy rows.

Once loaded, they get set to 0 if they're null: https://github.com/matrix-org/synapse/blob/b76f1a4d5f918def1f643910939b80e9e035e07f/synapse/storage/databases/main/keys.py#L75-L80

Maybe worth a comment explaining as much

**Review by reivilibre — APPROVED** [review]
otherwise LGTM

File: tests/crypto/test_keyring.py (hunk 1 above)
Comment (clokep) [reply tests/crypto/test_keyring.py:291]: 👍 That looks great, thanks!

**Pull request merged**